Add lint info to Async (#5956)
dart-lang/site-www: src/content/libraries/async/async-await.md

* Knowledge of [basic Dart syntax](/language).
* Some experience writing asynchronous code in another language.
* The [`discarded_futures`][] and [`unawaited_futures`][] lints enabled.

>   As with all lints, these lints aren't enabled by default.

Reviewer: I think this line can be removed since the default, or expected behavior, is that it's not enabled! Other than that, lgtm!

>   As with all lints, these lints aren't enabled by default.

Author: **issue:** Remove line.
```suggestion

```

</details>

## Which lints work for futures?

To catch common mistakes that arise while working with async and futures,
[enable](/tools/analysis#individual-rules) the following lints:

> * [discarded_futures][]
> * [unawaited_futures][]

Reviewer 2: ```suggestion
* [`discarded_futures`][]
* [`unawaited_futures`][]
```

> [discarded_futures]: /tools/linter-rules/discarded_futures
> [unawaited_futures]: /tools/linter-rules/unawaited_futures

Reviewer 2: ```suggestion
[`discarded_futures`]: /tools/linter-rules/discarded_futures
[`unawaited_futures`]: /tools/linter-rules/unawaited_futures
```

</details>

## Which lints work for futures?

> In real-world scenarios, enable the following two lints to help with
> futures.

Reviewer 2: "In real-world scenarios" doesn't add any meaning here. Instead I'd introduce the topic as:

```suggestion
To catch common mistakes that arise while working with async and futures,
[enable](/tools/analysis#individual-rules) the following lints:
```

* [discarded_futures][]
* [unawaited_futures][]

> By default, these lints aren't enabled.

Reviewer 2: No lints are enabled by default. This sentence is unnecessary.

> By default, these lints aren't enabled.

Author: **suggestion:** 
```suggestion

```
